models/dataset.go

```diff
@@ -307,24 +307,23 @@ func CreateContact(reader io.Reader) (*Contact, error) {
 
 // ValidateVersion checks the content of the version structure
 func ValidateVersion(version *Version) error {
-	var hasAssociation bool
 
 	switch version.State {
 	case "":
 		return errs.ErrVersionMissingState
 	case EditionConfirmedState:
-	case AssociatedState:
-		hasAssociation = true
 	case PublishedState:
-		hasAssociation = true
+		if version.CollectionID != "" {
```

Comment:
Could this cause issues for existing versions that have a collection ID's? or is the plan to fix the data so published versions will not have a collection ID?

Reply:
good question, I suspect the answer is yes: can @nshumoogum confirm?

Reply:
Yep as you both have realised a fix to the dataset documents in mongo is needed via a script. Previous data fix scripts are held [here](https://github.com/ONSdigital/dp-data-tools-go)